src/DBFTPlugin/ConsensusService.cs

```diff
@@ -56,21 +57,18 @@ internal ConsensusService(IActorRef localNode, IActorRef taskManager, IActorRef
             this.taskManager = taskManager;
             this.blockchain = blockchain;
             this.context = context;
+            node = localNode.Ask<LocalNode>(new LocalNode.GetInstance()).Result;
             Context.System.EventStream.Subscribe(Self, typeof(Blockchain.PersistCompleted));
             Context.System.EventStream.Subscribe(Self, typeof(Blockchain.RelayResult));
         }
 
         private bool AddTransaction(Transaction tx, bool verify)
         {
-            if (verify)
+            if (verify && !tx.Verify(DBFTPlugin.System.Settings, context.Snapshot, context.VerificationContext))
```

Comment:
Maybe `Verify()` should return `VerifyResult`?

Reply:
Please review again.